timeSeriesUtilities/combine_query_results_into_data_columns.py

```diff
@@ -0,0 +1,53 @@
+from connectors.core.connector import get_logger, ConnectorError
+from .constants import LOGGER_NAME
+logger = get_logger(LOGGER_NAME)
+
+
+def combine_query_results_into_data_columns(config, params):
+    dataCols = []
+    playbook_mode = params.get('playbookMode')
+    first_index = params.get('firstIndexToKeep')
+    time_buckets = params.get('queriedTimeBuckets')
+    query_results = params.get('queryResults')
+    input_columns = params.get('existingDataColumns')
+
+    # First remove any "too old" elements from the timestamp column, and then
+    # add on the new time buckets which were just queried
+    if input_columns and playbook_mode != "new_chart":
+        x_col = ['x'] + input_columns[0][first_index:-1] + [tb['start'] for tb in time_buckets]
+    else:
+        x_col = ['x'] + [tb['start'] for tb in time_buckets]
+
+    # Then remove all the too-old elements from the data lists
+    if input_columns and playbook_mode != "new_chart":
+        for column in input_columns[1:]:
+            newCol = [column[0]]
+            newCol += column[first_index:-1]
+            dataCols.append(newCol)
+
+    # Run through the results of the new queries
+    for qr in query_results:
```

Comment:
what if qr is null or None?

Reply:
Adding a default value of [] for query_results so that the for loop will be safely skipped over in that case